x/lscosmos/keeper/delegation_strategy.go

```diff
@@ -300,3 +300,123 @@ func (k Keeper) GetAllValidatorsState(ctx sdk.Context, denom string) (types.Allo
 	// returns the two updated lists
 	return updatedValSet, updatedDelegationState
 }
+
+// DelegateStrategy returns a list of types.ValAddressAmounts when provided with an input
+// types.DelegationState, types.AllowListedValidators and amount to be delegated
+func DelegateStrategy(valList types.AllowListedValidators, delegationState types.DelegationState, newAmount sdk.Coin) (types.ValAddressAmounts, error) {
+	// return nil if the new amount to be delegated is already zero
+	if newAmount.IsZero() {
+		return nil, nil
+	}
+
+	// get validators weight map
+	validatorWeightsMap := types.GetValidatorWeightsMap(valList)
+
+	// get total delegations post adding more to it
+	totalDelegationAfterAddingNewAmount := delegationState.TotalDelegations(newAmount.Denom).Add(newAmount).Amount
+
+	// get the difference, current and ideally delegated amounts for each validator
+	differenceCurrentAndIdealAmounts := GetDifferenceCurrentAndIdealAmounts(valList, delegationState, totalDelegationAfterAddingNewAmount)
+
+	// sort the difference, current and ideally delegated amounts for each validator based
+	// on the difference in descending order
+	sort.SliceStable(
+		differenceCurrentAndIdealAmounts,
+		func(i, j int) bool {
+			return differenceCurrentAndIdealAmounts[i].Diff.GT(differenceCurrentAndIdealAmounts[j].Diff)
+		},
+	)
+
+	// this is the first round of distribution to all the validators who are not
+	// yet matched to the ideal distribution
+	finalDistribution := make(map[string]sdk.Coin)
+	for _, i := range differenceCurrentAndIdealAmounts {
+
+		if i.Diff.GT(sdk.ZeroDec()) {
+			if newAmount.Amount.ToDec().LT(i.Diff.Abs()) && newAmount.Amount.Sub(newAmount.Amount).GTE(sdk.OneInt()) {
```

Comment:
`if i.Diff.GT(sdk.ZeroDec())` already checks if diff is positive, why need 
`i.Diff.Abs()`

Reply:
`newAmount.Amount.Sub(newAmount.Amount)`
shouldn't this always be zero?

Reply:
> `if i.Diff.GT(sdk.ZeroDec())` already checks if diff is positive, why need
> `i.Diff.Abs()`

Yes. Makes sense. Will remove this condition.
I think I did it in a haste.

Reply:
> `newAmount.Amount.Sub(newAmount.Amount)`
> shouldn't this always be zero?

Yeah. Need to subtract it for the next loop for further distribution.